pandas/core/frame.py

```diff
@@ -460,21 +460,29 @@ def __init__(
                 data = list(data)
             if len(data) > 0:
                 if is_list_like(data[0]) and getattr(data[0], "ndim", 1) == 1:
-                    if is_named_tuple(data[0]) and columns is None:
-                        columns = data[0]._fields
-                    arrays, columns = to_arrays(data, columns, dtype=dtype)
-                    columns = ensure_index(columns)
-
-                    # set the index
-                    if index is None:
-                        if isinstance(data[0], Series):
-                            index = get_names_from_index(data)
-                        elif isinstance(data[0], Categorical):
-                            index = ibase.default_index(len(data[0]))
-                        else:
-                            index = ibase.default_index(len(data))
-
-                    mgr = arrays_to_mgr(arrays, columns, index, columns, dtype=dtype)
+                    # try to infer that all elements are list-like as well
+                    try:
+                        if is_named_tuple(data[0]) and columns is None:
+                            columns = data[0]._fields
+                        arrays, columns = to_arrays(data, columns, dtype=dtype)
+                        columns = ensure_index(columns)
+
+                        # set the index
+                        if index is None:
+                            if isinstance(data[0], Series):
+                                index = get_names_from_index(data)
+                            elif isinstance(data[0], Categorical):
+                                index = ibase.default_index(len(data[0]))
+                            else:
+                                index = ibase.default_index(len(data))
+
+                        mgr = arrays_to_mgr(
+                            arrays, columns, index, columns, dtype=dtype
+                        )
+
+                    except TypeError:
```

Comment:
where is the TypeError actualy being raised? we dont' want to have very wide try/excepts they tend to hide errors.

Reply:
Its actually raised [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/internals/construction.py#L484). At that point, it is already inferred that it is a list of lists (which is not always true).

Debugging trace:
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../core/frame.py:465: in __init__
    arrays, columns = to_arrays(data, columns, dtype=dtype)
../../core/internals/construction.py:452: in to_arrays
    return _list_to_arrays(data, columns, coerce_float=coerce_float, dtype=dtype)
../../core/internals/construction.py:484: in _list_to_arrays
    content = list(lib.to_object_array(data).T)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: object of type 'int' has no len()
```

Reply:
We can also try to catch that specific exception with something like this?

```    
except TypeError as e:
    if not str(e).endswith("has no len()"):
        raise
    else:
        ....
```

Reply:
I don't want a try/except here at all, this just hides errors. you can do something more narrow in _list_to_arrays

Reply:
Okay thanks for the feedback! I'll have a look